meta/schemas/device.yaml

```diff
@@ -0,0 +1,332 @@
+
+  cpu:
+    $id: '#/properties/cpu'
+    title: CPU
+    type: object
+    required:
+      - model
+      - mhz
+      - cores
+    properties:
+        model:
+            $id: '#/properties/cpu/properties/model'
+            title: Model
+            type: string
+        cores:
+            $id: '#/properties/cpu/properties/cores'
+            title: The cores schema
+            type: integer
+            minimum: 1
+        mhz:
+            $id: '#/properties/cpu/properties/mhz'
+            title: MHz
+            type: integer
+            minimum: 1
+
+  detachable_antennas:
+    $id: '#/properties/detachable_antennas'
+    title: The detachable_antennas schema
+    type: number
+    minimum: 0
+
+  device_type:
+    $id: '#/properties/device_type'
+    $ref: file:device-type.yaml
+    title: Type
+    type: string
+
+  ethernet:
+    $id: '#/properties/ethernets'
+    title: Ethernets
+    type: array
+    items:
+      $ref: file:ethernet.yaml
+
+  usb:
+    $id: '#/properties/usb'
+    title: USB Ports
+    type: array
+    items:
+      $ref: file:usb.yaml
+
+  led:
+    $id: '#/properties/led'
+    title: LEDs
+    type: array
+    items:
+      $ref: file:led.yaml
+
+  jtag:
+    $id: '#/properties/jtag'
+    title: JTAG
+    type: boolean
+
+  auto_converted:
+    $id: '#/properties/auto_converted'
+    title: Automatically Converted
+    description: Migrated from Wiki data source
+    type: boolean
+
+  memory_mb:
+    $id: '#/properties/memory_mb'
+    title: Memory in MegaByte
+    type: array
+    minItems: 1
+    contains:
+      type: number
+
+  model:
+    $id: '#/properties/model'
+    title: Model
+    type: string
+
+  supported_since_commit:
+    $id: '#/properties/supported_since_commit'
+    title: Supported Since Git Commit
+    type: string
+    pattern: "^[a-zA-Z0-9]+$"
+    minLength: 6
+    maxLength: 40
+
+  variant:
+    $id: '#/properties/variant'
+    title: Variant
+    type: string
+
+  outdoor:
+    $id: '#/properties/outdoor'
+    title: Outdoor
+    type: boolean
+
+  power_supply:
+    $id: '#/properties/power_supply'
+    $ref: file:power_supply.yaml
+
+  serial:
+    $id: '#/properties/serial'
+    title: Serial
+    type: string
+    enum:
+      - unknown
+      - unknown_params
+      - none
+      - 9600/8N1
+      - 19200/8N1
+      - 38400/8N1
+      - 57600/8N1
+      - 115200/8N1
+      - 1500000/8N1
+
+  sata_ports:
+    $id: '#/properties/sata_ports'
+    title: Sata Ports
+    type: integer
+    minimum: 0
+
+  gpio_pins:
+    $id: '#/properties/gpio_pins'
+    title: GPIO Pins
+    type: integer
+    minimum: 0
+
+  storage:
+    $id: '#/properties/storage'
+    title: Storage
+    type: array
+    items:
+      $ref: file:storage.yaml
+    minItems: 1
+
+  target:
+    $id: '#/properties/target'
+    type: array
+    items:
+      $ref: file:target.yaml
+    minItems: 1
+
+  vendor:
+    $id: '#/properties/vendor'
+    $ref: file:vendor.yaml
+
+  vlan:
```

Comment:
This is probably going to be more complex. This is switch property and should be modeled as that. Device can have 1-N switches, they might be of different hardware/type and thus features etc.